src/MobileRedirection.php

```diff
@@ -64,15 +64,20 @@ public function register() {
 		add_filter( 'amp_default_options', [ $this, 'filter_default_options' ] );
 		add_filter( 'amp_options_updating', [ $this, 'sanitize_options' ], 10, 2 );
 
-		$is_mobile_redirect_enabled = AMP_Options_Manager::get_option( Option::MOBILE_REDIRECT );
+		$is_amp_canonical           = amp_is_canonical();
 		$sandboxing_level           = amp_get_sandboxing_level();
+		$is_mobile_redirect_enabled = AMP_Options_Manager::get_option( Option::MOBILE_REDIRECT );
 
 		// Add alternative link if mobile redirection is enabled or sandboxing level is set to loose or moderate.
-		if ( ! amp_is_canonical() && ( $is_mobile_redirect_enabled || ( 1 === $sandboxing_level || 2 === $sandboxing_level ) ) ) {
+		if ( ! $is_amp_canonical && ( $is_mobile_redirect_enabled || ( 1 === $sandboxing_level || 2 === $sandboxing_level ) ) ) {
 			add_action( 'wp_head', [ $this, 'add_mobile_alternative_link' ] );
 		}
 
-		if ( $is_mobile_redirect_enabled && ! amp_is_canonical() ) {
+		if ( ! $is_mobile_redirect_enabled && ! $is_amp_canonical ) {
+			add_action( 'template_redirect', [ $this, 'maybe_add_mobile_switcher_link' ] );
+		}
+
+		if ( $is_mobile_redirect_enabled && ! $is_amp_canonical ) {
```

Comment:
Since both of these conditions require `! $is_amp_canonical`, would it make sense to combine them to be like:

```php
if ( ! $is_amp_canonical ) {
    if ( ! $is_mobile_redirect_enabled ) {
        // ...
    } else {
        // ...
    }
}
```

This should simplify the conditions and make it easier to read.

Reply:
Actually, all three conditionals use `! $is_amp_canonical` so makes sense to group all of them in one.